packages/react-ui-core/src/Gmap/Gmap.js

```diff
@@ -1,86 +1,162 @@
 import React, { PureComponent } from 'react'
 import PropTypes from 'prop-types'
-import autobind from 'autobind-decorator'
-import { parseArgs } from '@rentpath/react-ui-utils'
-import GmapBase from './GmapBase'
+import themed from 'react-themed'
+import classnames from 'classnames'
+import startsWith from 'lodash/startsWith'
+import pickBy from 'lodash/pickBy'
+import { GmapInteraction } from './GmapInteraction'
+import withGoogleScript from './withGoogleScript'
 
-const API_BASE_URL = 'https://maps.googleapis.com/maps/api/js'
+const DEFAULT_ZOOM = 8
+const DEFAULT_CENTER = { // Atlanta, GA
+  lat: 33.7490,
+  lng: -84.3880,
+}
+
+const EVENTS = {
+  onBoundsChanged: 'bounds_changed',
+  onCenterChanged: 'center_changed',
+  onClick: 'click',
+  onDoubleClick: 'dblclick',
+  onDrag: 'drag',
+  onDragEnd: 'dragend',
+  onDragStart: 'dragstart',
+  onHeadingChnaged: 'heading_changed',
+  onIdle: 'idle',
+  onMaptypeIdChanged: 'maptypeid_changed',
+  onMouseMove: 'mousemove',
+  onMouseOut: 'mouseout',
+  onMouseOver: 'mouseover',
+  onProjectionChanged: 'projection_changed',
+  onResize: 'resize',
+  onRightclick: 'rightclick',
+  onTilesLoaded: 'tilesloaded',
+  onTiltChanged: 'tilt_changed',
+  onZoomChanged: 'zoom_changed',
+}
 
-export default class Gmap extends PureComponent {
+const EVENT_NAMES = Object.keys(EVENTS)
+
+@themed(/^Gmap/, { pure: true })
+
+export class Gmap extends PureComponent {
   static propTypes = {
-    apiKey: PropTypes.string.isRequired,
-    libraries: PropTypes.arrayOf(
-      PropTypes.string,
-    ),
-    spinner: PropTypes.oneOfType([
-      PropTypes.func,
-      PropTypes.node,
-      PropTypes.object,
-    ]),
+    theme: PropTypes.object,
+    className: PropTypes.string,
+    children: PropTypes.node,
+    center: PropTypes.shape({
+      lat: PropTypes.number,
+      lng: PropTypes.number,
+    }),
+    zoom: PropTypes.number,
+    clickableIcons: PropTypes.bool,
+    mapControls: PropTypes.shape({
+      fullscreenControl: PropTypes.bool,
+      mapTypeControl: PropTypes.bool,
+      zoomControl: PropTypes.bool,
+      streetViewControl: PropTypes.bool,
+    }),
   }
 
   static defaultProps = {
-    libraries: [],
+    theme: {},
+    zoom: DEFAULT_ZOOM,
+    clickableIcons: false,
+    mapControls: {
+      fullscreenControl: false,
+      streetViewControl: false,
```

Comment:
what about `mapTypeControl` and `zoomControl` defaults?

Reply:
We can add them, I didn't want to be overly aggressive when adding these as defaults.